src/main/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestworker/services/GeneratePdfService.kt

```diff
@@ -159,12 +159,19 @@ class GeneratePdfService {
   }
 
   fun preProcessData(input: Any?): Any? {
+    if (input is List<*> && input.isEmpty()) {
```

Comment:
They're returned as ArrayList<> rather than List<>.

Also I feel like this would be neater if we removed this part and instead added "&& !input.isEmpty()" to the other if statement, so it looks more like:

- If it's a map, process each key and recursively iterate through each value
- If it's a list (and not empty), recursively iterate through each value
- If it's an empty list OR null OR the string "null", replace it
- Otherwise, return it.

I think this logic flows better and means we don't have some processing up here, some processing in the maps bit, etc.